### PR TITLE
fix multiple form embedded issue in site page

### PR DESCRIPTION
**Pull request by amit-mnnit**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes and the overall coverage did not decrease.
- [ ] All unit tests pass on CircleCi.
- [ ] I ran all tests locally and they pass.

### CODE REVIEW

**Review by sakshi-arora1 — COMMENTED** [review]
This is the check I was talking about. 
https://github.com/adobe/aem-core-forms-components/blob/master/ui.frontend/src/view/FormContainer.js#L93

This should prevent fields getting added to other formcontainer other than its own.

File: ui.frontend/src/utils.js [hunk 1]
```diff
@@ -117,8 +117,10 @@ class Utils {
     static #createFormContainerFields(fieldElements, fieldCreator, formContainer) {
         for (let i = 0; i < fieldElements.length; i++) {
             const elementId = fieldElements[i].id;
+            const fieldPath = fieldElements[i].getAttribute('data-cmp-adaptiveformcontainer-path');
+            const formPath = formContainer.getFormElement().getAttribute('data-cmp-path');
             //check if field is already created, to avoid creating a new field for same id
-            if (formContainer.getField(elementId) == null) {
+            if (formContainer.getField(elementId) == null && fieldPath == formPath) {
```
Comment (sakshi-arora1) [inline ui.frontend/src/utils.js:123]: this is another preventive check, but you need to check repeatability with this and a test case for same.

File: ui.frontend/src/utils.js [hunk 2]
```diff
@@ -117,8 +117,10 @@ class Utils {
     static #createFormContainerFields(fieldElements, fieldCreator, formContainer) {
         for (let i = 0; i < fieldElements.length; i++) {
             const elementId = fieldElements[i].id;
+            const fieldPath = fieldElements[i].getAttribute('data-cmp-adaptiveformcontainer-path');
```
Comment (rismehta) [inline ui.frontend/src/utils.js:120]: Can you rename this to formPathInField

File: ui.frontend/src/utils.js (hunk 1 above)
Comment (rismehta) [reply ui.frontend/src/utils.js:123]: Even repeatable element would have these attributes right ?

File: ui.tests/test-module/specs/aemEmbedContainer/aemEmbedContainer.runtime.spec.js [hunk 3]
```diff
@@ -125,4 +125,81 @@ describe("Sites with Aem Embed Container", () => {
             })
         })
     })
+
+    context('embed multiple form in site ', function () {
```
Comment (rismehta) [inline ui.tests/test-module/specs/aemEmbedContainer/aemEmbedContainer.runtime.spec.js:129]: Please create a separate spec for this. This is not part of aem embed container.

File: ui.frontend/src/utils.js (hunk 2 above)
Comment (amit-mnnit) [reply ui.frontend/src/utils.js]: done

File: ui.tests/test-module/specs/aemEmbedContainer/aemEmbedContainer.runtime.spec.js (hunk 3 above)
Comment (amit-mnnit) [reply ui.tests/test-module/specs/aemEmbedContainer/aemEmbedContainer.runtime.spec.js]: done

File: ui.frontend/src/utils.js (hunk 1 above)
Comment (amit-mnnit) [reply ui.frontend/src/utils.js]: done

**Comment by rismehta** [comment]
Any comments can be taken up iteratively, this LGTM.

**Pull request merged**